src/main/java/com/aws/greengrass/deployment/ConnectivityValidator.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.deployment;
+
+import com.aws.greengrass.config.Topic;
+import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.deployment.errorcode.DeploymentErrorCode;
+import com.aws.greengrass.deployment.exceptions.ComponentConfigurationValidationException;
+import com.aws.greengrass.deployment.model.DeploymentResult;
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+import com.aws.greengrass.mqttclient.MqttClient;
+import com.aws.greengrass.util.Coerce;
+import com.aws.greengrass.util.GreengrassServiceClientFactory;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import software.amazon.awssdk.crt.mqtt.MqttClientConnection;
+import software.amazon.awssdk.crt.mqtt.MqttException;
+import software.amazon.awssdk.http.SdkHttpClient;
+import software.amazon.awssdk.iot.AwsIotMqttConnectionBuilder;
+import software.amazon.awssdk.services.greengrassv2data.GreengrassV2DataClient;
+import software.amazon.awssdk.services.greengrassv2data.model.ListThingGroupsForCoreDeviceRequest;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class ConnectivityValidator implements AutoCloseable {
+    private static final Logger logger = LogManager.getLogger(ConnectivityValidator.class);
+    private static final int maxRetries = 3;
+    public static final String clientIdSuffix = "#validation";
+    private static final int mqttWaitTimeSeconds = 2;
+    private static final int httpWaitTimeSeconds = 5;
```

Comment:
configurable.

Reply:
This is the wait time between retries or do we consider the timeouts set by the customer configs to be enough and just immediately retry?

Reply:
Waiting seems unlikely to resolve anything, especially only waiting a couple of seconds. I'm not sure that there's any value in having retries actually.

Reply:
I'll remove the wait time since waiting a couple of seconds isn't likely to impact anything

But wouldn't we want to at least retry a few times in case of an unstable network?

Reply:
Retrying, without a wait, is very unlikely to help anything.